sdk/storage/azure-storage-common/inc/azure/storage/common/paged_response.hpp

```diff
@@ -30,7 +30,8 @@ template <class Derived> class PagedResponse {
       m_hasMorePages = false;
       return;
     }
-    CurrentPageToken = NextPageToken;
```

Comment:
OK, so removing this assignment helps. Where does it get assigned then?
Are you sayin the contract is for all client types that implement `PagedRespons` must set it in their implementation. For example, it needs to be set within `ListBlobsByHierarchy`?

And the same is applied to all three public fields on `PagedResponse` (including RawResponse and NextPageToken)?

Reply:
The line `CurrentPageToken = NextPageToken;` which I just removed doesn't have any effect at all. It was redundant.

If you look at the implementation of `OnNextPage` like this one

https://github.com/Azure/azure-sdk-for-cpp/blob/5570a0ae4f358ffa71bf94b75763ae9235b1a8c7/sdk/storage/azure-storage-blobs/src/blob_responses.cpp#L67

the whole paged response including the derived class and its base class is overwritten.

Reply:
> the whole paged response including the derived class and its base class is overwritten.

Makes sense. This implementation requirement, and other aspects of the `PagedResponse<T>` that client library authors need to know about should be included as part of the C++ guideline (and in the API reference docs). FYI, @RickWinter